onnxruntime/core/providers/cuda/tensor/tile_impl.cu

```diff
@@ -75,43 +100,95 @@ __global__ void _TileMemcpyKernel(const T* input_data, T* output_data, const CUD
   }
 }
 
+template <typename T>
+size_t GetVectorizedSize(size_t num_input_elements, size_t num_elements_per_batch, uint64_t address_input,
+                         uint64_t address_output, CUDA_LONG& N, int& blocksPerGrid) {
+  constexpr int vec4_alignment = std::alignment_of<aligned_vector<T, 4>>::value;
+  constexpr int vec2_alignment = std::alignment_of<aligned_vector<T, 2>>::value;
+  N = static_cast<CUDA_LONG>(num_input_elements);
+  size_t vectorized_size = 1;
+  if (num_elements_per_batch % 4 == 0 & address_input % vec4_alignment == 0 && address_output % vec4_alignment == 0) {
+    N /= 4;
+    vectorized_size = 4;
+  } else if (num_elements_per_batch % 2 == 0 && address_input % vec2_alignment == 0 &&
+             address_output % vec2_alignment == 0) {
+    N /= 2;
+    vectorized_size = 2;
+  }
+  blocksPerGrid = CeilDiv(N, num_threads_per_block);
+  return vectorized_size;
+}
+
 template <typename T>
 void TileMemcpyImpl(cudaStream_t stream, const T* input_data, T* output_data, const size_t num_input_elements,
                     const size_t repeats) {
-  using Vec4T = aligned_vector<T, 4>;
-  using Vec2T = aligned_vector<T, 2>;
-  constexpr int vec4_alignment = std::alignment_of<Vec4T>::value;
-  constexpr int vec2_alignment = std::alignment_of<Vec2T>::value;
-  uint64_t address_input = reinterpret_cast<uint64_t>(input_data);
-  uint64_t address_output = reinterpret_cast<uint64_t>(output_data);
-  CUDA_LONG N = static_cast<CUDA_LONG>(num_input_elements);
-  if (num_input_elements % 4 == 0 && address_input % vec4_alignment == 0 && address_output % vec4_alignment == 0) {
-    N /= 4;
-    int blocksPerGrid = CeilDiv(N, num_threads_per_block);
-    _TileMemcpyKernel<<<blocksPerGrid, num_threads_per_block, 0, stream>>>(
+  // If the block number from input size is too small to fill all streaming multiprocessors,
```

Comment:
nit: Feels like the comment is incomplete. Maybe you would like to finish it by saying, "in this case we want to use the output based kernel" or something to that effect ?